src/transport_ofi.c

```diff
@@ -603,24 +610,44 @@ static inline int exchange_and_av_insert(struct fabric_info *info)
 	return ret;
     }
 
+    full_ep_len = full_ep_len + epnamelen;
+    epnamebuff = malloc(full_ep_len);
+
     alladdrs = malloc(info->npes * epnamelen);
     if(alladdrs==NULL){
 	OFI_ERRMSG("alladdrs is NULL\n");
 	return ret;
     }
 
-    ret = shmem_runtime_put("OFI", epname, epnamelen);
+#ifdef USE_ON_NODE_COMMS
+    gethostname(&epname[epnamelen], pidlen);
+#endif
+
+    ret = shmem_runtime_put("OFI", epname, full_ep_len);
     shmem_runtime_exchange();
     shmem_runtime_barrier();
 
     for(i=0; i<info->npes; i++)
     {
-	    void *tgt = alladdrs + i*epnamelen;
+	    adrlen = i*epnamelen;
+
 	    if (i == rank) {
-		    memcpy(tgt, epname, epnamelen);
+		    memcpy(epnamebuff, epname, full_ep_len);
```

Comment:
actually epname is either as large or larger than full_ep_len. At first full_ep_len is of a larger than needed size including the predicted size of hostname. After fi_getname, this readjusts the epnamelen to a likely much smaller value. For instances in sockets its around ~16. At this point epnamebuff is created with the adjusted epnamelen size + hostname_max size. epname is still at its original size so by the time you hit this memcpy you have full_ep_len = sizeof(epnamebuff) which is >=  sizeof(epname)